STYLE_GUIDE.md

```diff
@@ -0,0 +1,156 @@
+# Go Coding Style Guide
+
+In order to keep our code looking good with lots of programmers working on it, it helps to have a "style guide", so all
+the code generally looks quite similar. This doesn't mean there is only one "right way" to write code, or even that this
+standard is better than your style.  But if we agree to a number of stylistic practices, it makes it much easier to read
+and modify new code. Please feel free to make suggestions if there's something you would like to add or modify.
+
+Please, before going through the rest of the document, have a read over the [syntactical style](https://github.com/uber-go/guide/blob/master/style.md)
+
+If you wish to gain some further understanding, we recommend having a read on how to code [effective go](https://golang.org/doc/effective_go.html)
+
+
+## Code Structure
+
+Perhaps more key for code readability than good commenting is having the right structure. As a rule of thumb, try to write
+in a logical order of importance, taking a little time to think how to order and divide the code such that someone could
+scroll down and understand the functionality of it just as well as you do. A loose example of such order would be:
+* Constants and global variables
+* Main Struct
+* Options (only if they are seen as critical to the struct else they should be placed in another file)
+* Initialisation / Start and stop of the service
+* Msgs/Events
+* Public Functions (In order of most important)
+* Private/helper functions
+* Auxiliary structs and function (can also be above private functions or in a separate file)
+
+## General
+
+ * Use `gofmt` (or `goimport`) to format all code upon saving it.  (If you use VIM, check out vim-go).
+ * Use a linter (see below) and generally try to keep the linter happy (where it makes sense).
+ * Think about documentation, and try to leave godoc comments, when it will help new developers.
+ * Every package should have a high level doc.go file to describe the purpose of that package, its main functions, and any other relevant information.
+ * `TODO` should not be used. If important enough should be recorded as an issue.
+ * `BUG` / `FIXME` should be used sparingly to guide future developers on some of the vulnerabilities of the code.
+ * `XXX` can be used in work-in-progress (prefixed with "WIP:" on github) branches but they must be removed before approving a PR.
+ * Applications (e.g. clis/servers) *should* panic on unexpected unrecoverable errors and print a stack trace.
+
+## Comments
+
+ * Use a space after comment deliminter (ex. `// your comment`).
+ * Many comments are not sentences, these should begin with a lower case letter and end without a period.
+ * The first letter of sentences in comments are capitalized and ends with a period.
+
+## Linters
+
+These must be applied to all (go) repos.
```

Comment:
Would be nice to add instructions on running these linters too! (You could add a TODO and do it in a follow up.)

Reply:
Do you mean including something like:
After installing, use `golangci-lint run` in the directory you would like to check

The linters have links to their respective repo's if people want to take a deeper look into them

Reply:
Are you okay with this, or do you still want more instructions?